### PR TITLE
Google maps

### PR DESCRIPTION
**Pull request by AthenaPK**
Closes issue #8, Google Maps.

### CODE REVIEW

**Review by bluedrack — COMMENTED** [review]
SGTM but maybe don't modify the MainActivity to make the merge more easy with the other bootcamp.

**Review by bluedrack — DISMISSED** [review]
LGTM

**Comment by AthenaPK** [comment]
Closes #23 but depends on #24, so do not merge before #24 is complete.

File: app/src/main/java/com/github/studydistractor/sdp/FireBaseProcrastinationActivityService.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+
+class FireBaseProcrastinationActivityService : ProcrastinationActivityService{
```
Comment (aizensoosuke) [inline app/src/main/java/com/github/studydistractor/sdp/FireBaseProcrastinationActivityService.kt:14]: This should probably be called FirebaseProcrastinationService, it should not be bound to an activity.

Comment (aizensoosuke) [reply app/src/main/java/com/github/studydistractor/sdp/FireBaseProcrastinationActivityService.kt:14]: As you said in willy's review

File: app/src/main/java/com/github/studydistractor/sdp/MapsActivity.kt [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+     * Displays the places on the map and launches the distraction activity when clicking on the marker
+     */
+    fun displayPlaces() {
+        val sat = LatLng(46.520544, 6.567825)
+//        TODO: set marker style according to our color scheme and perhaps according to the type of place
+        map?.addMarker(
+            MarkerOptions().position(sat).title("Satellite")
+                .icon(BitmapDescriptorFactory.defaultMarker(BitmapDescriptorFactory.HUE_MAGENTA))
+        )
+//        TODO: which activity to launch when clicking on the marker? For now, it launches the main activity, because it's the only one, but it should launch the distraction activity
```
Comment (aizensoosuke) [inline app/src/main/java/com/github/studydistractor/sdp/MapsActivity.kt:71]: Done

File: app/src/main/java/com/github/studydistractor/sdp/ProcrastinationActivity.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.github.studydistractor.sdp
+
+import android.os.Parcel
+import android.os.Parcelable
+
+data class ProcrastinationActivity(
+    val name: String? = null,
+    val description: String? = null) :
+    Parcelable {
```
Comment (aizensoosuke) [inline app/src/main/java/com/github/studydistractor/sdp/ProcrastinationActivity.kt:9]: This should not be used in future sprints. We want to manage state with compose (see [this](https://developer.android.com/jetpack/compose/state))

**Review by aizensoosuke — APPROVED** [review]
All good thank you!

File: app/src/main/java/com/github/studydistractor/sdp/ProcrastinationActivity.kt (hunk 3 above)
Comment (aizensoosuke) [reply app/src/main/java/com/github/studydistractor/sdp/ProcrastinationActivity.kt:9]: Converted to issue : #36

File: app/src/main/java/com/github/studydistractor/sdp/MapsActivity.kt (hunk 2 above)
Comment (AthenaPK) [reply app/src/main/java/com/github/studydistractor/sdp/MapsActivity.kt]: Good point, thanks!

**Review by aizensoosuke — APPROVED** [review]
Thanks for the changes

**Pull request merged**